pkg/apis/policy/types.go

```diff
@@ -42,8 +42,37 @@ type PodDisruptionBudgetSpec struct {
 	// by specifying 0. This is a mutually exclusive setting with "minAvailable".
 	// +optional
 	MaxUnavailable *intstr.IntOrString
+
+	// PodHealthyPolicy defines the criteria for when the disruption controller
+	// should consider a pod to be healthy.
+	// If no policy is specified, the legacy behavior will be used. It means
+	// only pods that are Running and Ready will be considered when the disruption
+	// controller computes "disruptionsAllowed", but all pods in the Running phase
+	// will be subject to the PDB on eviction.
+	// +optional
+	PodHealthyPolicy PodHealthyPolicy
```

Comment:
Add information this is alpha-level field and that feature flag needs to be enabled for it to work: similarly to https://github.com/kubernetes/kubernetes/blob/a83c97941b63e800805378124a5fab08a051e4d4/pkg/apis/batch/types.go#L246-L251